docs/tapiro/installation.md

```diff
@@ -0,0 +1,100 @@
+---
+id: installation
+title: Installation
+---
+
+`tapiro` can be installed as an sbt-plugin.
+
+`sbt-tapiro` is an Sbt plugin that uses `tapiro` to generate http/json routes parsing scala traits definitions.
+
+## Installation
+
+To start using `sbt-tapiro` simply add this line in `project/plugins.sbt`
+
+```scala
+addSbtPlugin("io.buildo" %% "sbt-tapiro" % "@SBT_TAPIRO_STABLE_VERSION@")
+```
+
+### Snapshot releases
+
+We publish a snapshot version on every merge on master.
+
+The latest snapshot version is `@SBT_BUILDO_SNAPSHOT_VERSION@` and you can use
+it to try the latest unreleased features. For example:
+
+```scala
+addSbtPlugin("io.buildo" %% "sbt-tapiro" % "@SBT_TAPIRO_SNAPSHOT_VERSION@")
+resolvers += Resolver.sonatypeRepo("snapshots")
+```
+
+## Plugin
+
+To use the code generator, you need to add this to your `build.sbt`.
+
+```scala
+lazy val application = project
+  .settings(
+    libraryDependencies ++= applicationDependencies ++ tapiroDependencies,
+    tapiro / tapiroRoutesPaths := List("[path to routes]"),
+    tapiro / tapiroModelsPaths := List("[path to models]"),
+    tapiro / tapiroOutputPath := "[path to endpoints]",
+    tapiro / tapiroEndpointsPackages := List("[package]", "[subpackage]"),
+    tapiro / tapiroServer := Server.AkkaHttp, //or Server.Http4s
+  )
+  .enablePlugins(SbtTapiro)
+```
+
+You can now run it with `sbt application/tapiro`.
+
+In order to use `tapiro` you also need to add the macro paradise plugin (for
+Scala 2.12) or to enable the `-Ymacro-annotations` compiler flag (for Scala
+2.13).
+
+> 💡**PROTIP**: This is done automatically for you if you use `sbt-buildo`
+
+```scala
+// Scala 2.12
+addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.1" cross CrossVersion.full)
+
+// Scala 2.13
+scalacOptions += "-Ymacro-annotations"
+```
+
+## Dependencies
+
+The generated code comes with library dependencies.
+
+In case akka-http version is used:
+```scala
+val V = new {
+  val circe = "0.9.1"
+  val tapir = "0.12.19"
+  val akkaHttp = "10.1.11"
```

Comment:
Queste versioni si possono iniettare dinamicamente

Reply:
In realtà no, rispecchiano versione usata da tapiro, ma non sono dipendenze di tapiro.

Reply:
O meglio, si possono cmq iniettare dinamicamente, però staremmo aggiungendo dipendenza solo per iniettarle. Potrebbe essere un buon modo per tenerne traccia, però è un pelo più intricato.

Reply:
forse si può giocare con le dipendenze opzionali, comunque sicuramente meglio un altro issue

Reply:
Done